app/scripts/views/settings/security.js

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const $ = require('jquery');
+  const AvatarMixin = require('../mixins/avatar-mixin');
+  const AuthErrors = require('lib/auth-errors');
+  const BaseView = require('../base');
+  const Cocktail = require('cocktail');
+  const FloatingPlaceholderMixin = require('../mixins/floating-placeholder-mixin');
+  const FormView = require('../form');
+  const SettingsPanelMixin = require('../mixins/settings-panel-mixin');
+  const SearchParamMixin = require('../../lib/search-param-mixin');
+  const Template = require('templates/settings/security.mustache');
+  const preventDefaultThen = require('../base').preventDefaultThen;
+  const showProgressIndicator = require('../decorators/progress_indicator');
+
+  var t = BaseView.t;
+
+  const CODE_INPUT_SELECTOR = 'input.totp-code';
+  const CODE_REFRESH_SELECTOR = 'button.settings-button.totp-refresh';
+  const CODE_REFRESH_DELAYMS = 350;
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'security',
+    viewName: 'settings.security',
+
+    events: {
+      'click .show-code-link': preventDefaultThen('_showCode'),
+      'click .totp-cancel': preventDefaultThen('cancel'),
+      'click .totp-confirm-code': preventDefaultThen('confirmCode'),
+      'click .totp-create': preventDefaultThen('createToken'),
+      'click .totp-delete': preventDefaultThen('deleteToken'),
+      'click .totp-refresh': preventDefaultThen('refresh'),
+    },
+
+    _checkTokenExists() {
+      const account = this.getSignedInAccount();
+      return account.checkTotpTokenExists()
+        .then((result) => {
+          this._hasToken = result.exists;
+        });
+    },
+
+    _santizeCode (code) {
+      // Remove spaces and `-`
+      return code.replace(/[- ]*/g, '');
+    },
+
+    _showQrCode() {
+      this.$('#totp').removeClass('hidden');
+    },
+
+    _showStatus() {
+      this.$('.security-list').removeClass('hidden');
+    },
+
+    _hideStatus() {
+      this.$('.security-list').addClass('hidden');
+    },
+
+    _showCode() {
+      $('.manual-code').removeClass('hidden');
+      $('.show-code-link').addClass('hidden');
+    },
+
+    _getFormattedCode(code) {
+      // Insert spaces every 4 characters
+      return code.replace(/(\w{4})/g, '$1 ');
```

Comment:
nit: I am worried that adding spaces might annoy people if they copy paste and use some cli totp app that will mess up with actual spaces. However 1password seemed to able to figure out the spaces 

*shrugs*

Reply:
I tested this with Authenticator and it worked as expected. It was added to that the code was more readable.

Reply:
If there is no easy way CSS can space out every 4th character then no worries